sympy/solvers/solveset.py

```diff
@@ -548,6 +528,9 @@ def _solveset(f, symbol, domain, _check=False):
         a, h = f.as_independent(symbol)
         m, h = h.as_independent(symbol, as_Add=False)
         f = a/m + h  # XXX condition `m != 0` should be added to soln
+    numer, denom = f.as_numer_denom()
```

Comment:
Here is a case that works with the change above because the singularity ends up providing an infinity in the denominator that then makes the expression go to 0 (i.e. 0/oo -> 0):

```
>>> eq = x/(1/x - 1)
>>> n,d=eq.as_numer_denom()
>>> sol = solveset(n,x,S.Reals) - solveset(d,x,S.Reals)
>>> sol
{0}
```

But if you replace that `- 1` with `- y` the solution of 0 will not work unless y is finite since, for example, `1/0 - oo` is  not defined. I think that is the motivation for requiring that the factors be known as finite.